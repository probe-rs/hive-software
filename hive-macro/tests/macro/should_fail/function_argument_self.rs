#[hive_macro::hive_test]
fn test_function(self, handle: String) {}

fn main() {}
