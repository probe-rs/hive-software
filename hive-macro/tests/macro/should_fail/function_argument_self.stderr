error: Function should not take self as argument.

         = help: Hive testfunctions do not use self, see the example below.
         = note: example: fn my_fancy_test(test_channel: &mut dyn TestChannelHandle, session: &mut Session, target_info: &HiveTargetInfo);

 --> tests/macro/should_fail/function_argument_self.rs:2:18
  |
2 | fn test_function(self, handle: String, info: String) {}
  |                  ^^^^
